python/mspasspy/io/distributed.py

```diff
@@ -0,0 +1,1322 @@
+                                        k, type(md[k])
+                                    )
+                                )
+
+        elif mode == "pedantic":
+            for k in md:
+                if read_metadata_schema.is_defined(k):
+                    if not isinstance(md[k], read_metadata_schema.type(k)):
+                        fatal_keys.append(k)
+                        is_dead = True
+                        log_error_msg.append(
+                            "pedantic mode: {} has type {}, forbidden by definition".format(
+                                k, type(md[k])
+                            )
+                        )
+
+        # 1.4 create a mspass object by passing MetaData
+        # if not changing the fatal key values, runtime error in construct a mspass object
+        for k in fatal_keys:
+            if read_metadata_schema.type(k) is str:
+                md[k] = ""
+            elif read_metadata_schema.type(k) is int:
+                md[k] = 0
+            elif read_metadata_schema.type(k) is float:
+                md[k] = 0.0
+            elif read_metadata_schema.type(k) is bool:
+                md[k] = False
+            elif read_metadata_schema.type(k) is dict:
+                md[k] = {}
+            elif read_metadata_schema.type(k) is list:
+                md[k] = []
+            elif read_metadata_schema.type(k) is bytes:
+                md[k] = b"\x00"
+            else:
+                md[k] = None
+
+        # init a ProcessingHistory to store history
+        h = ProcessingHistory()
+
+        # not continue step 2 & 3 if the mspass object is dead
+        if is_dead:
+            md["is_dead"] = True  # mspass_object.kill()
+            for msg in log_error_msg:
+                h.elog.log_error("read_data", msg, ErrorSeverity.Invalid)
+        else:
+            md["is_dead"] = False  # mspass_object.live = True
+
+            # 3.load history
+            if load_history:
+                history_obj_id_name = (
+                    db.database_schema.default_name("history_object") + "_id"
+                )
+                if history_obj_id_name in object_doc:
+                    # Load (in place) the processing history into h.
+                    history_object_id = object_doc[history_obj_id_name]
+                    # get the atomic type of the mspass object
+                    if object_type is TimeSeries:
+                        atomic_type = AtomicType.TIMESERIES
+                    else:
+                        atomic_type = AtomicType.SEISMOGRAM
+                    if not collection:
+                        collection = db.database_schema.default_name("history_object")
+                    # load history if set True
+                    res = db[collection].find_one({"_id": history_object_id})
+                    # retrieve_history_record
+                    if retrieve_history_record:
+                        h = pickle.loads(res["processing_history"])
```

Comment:
Never use a single letter variable name in any code. This will make the code hard to read and maintain. Long and explicit words are always preferred. So, instead of calling `h`, this should be something like `processing_history_record`